data/org.freedesktop.impl.portal.GlobalShortcuts.xml

```diff
@@ -0,0 +1,151 @@
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public
+ License along with this library. If not, see <http://www.gnu.org/licenses/>.
+
+ Author: Aleix Pol Gonzalez <aleixpol@kde.org>
+-->
+
+<node name="/" xmlns:doc="http://www.freedesktop.org/dbus/1.0/doc.dtd">
+  <!--
+      org.freedesktop.impl.portal.GlobalShortcut:
+      @short_description: GlobalShortcut portal backend interface
+
+      This this portal lets applications register global shortcuts so they can
+      act regardless of the system state upon an input event.
+  -->
+  <interface name="org.freedesktop.impl.portal.GlobalShortcuts">
+    <!--
+        CreateSession:
+        @handle: Object path for the #org.freedesktop.impl.portal.Request object representing this call
+        @session_handle: Object path for the #org.freedesktop.impl.portal.Session object representing the session being created
+        @app_id: App id of the application
+        @results: Vardict with the results of the call
+
+        Create a global shortcuts session.
+
+        The following results get returned via the #org.freedesktop.portal.Request::Response signal:
+        <variablelist>
+          <varlistentry>
+            <term>session_id s</term>
+            <listitem><para>
+              The session id. A string representing the created global shortcuts session.
+            </para></listitem>
+          </varlistentry>
+          <varlistentry>
+            <term>shortcuts a{sv}</term>
+            <listitem><para>
+              The ids of the supported shortcuts mapped to a user-readable
+              description of what it does.
+            </para></listitem>
+          </varlistentry>
+      </variablelist>
+    -->
+    <method name="CreateSession">
+      <arg type="o" name="handle" direction="in"/>
+      <arg type="o" name="session_handle" direction="in"/>
+      <arg type="s" name="app_id" direction="in"/>
+      <arg type="a{sv}" name="results" direction="out"/>
+    </method>
+
+    <!--
+        ListShortcuts:
+        @handle: Object path for the #org.freedesktop.impl.portal.Request object representing this call
+        @session_handle: Object path for the #org.freedesktop.impl.portal.Session object representing the session
+        @results: Vardict with the results of the call
+
+        List the shortcuts registered in the @p handle session.
+
+        The following results get returned via the #org.freedesktop.portal.Request::Response signal:
+        <variablelist>
+          <varlistentry>
+            <term>names a{sv}</term>
+            <listitem><para>
+              The different shortcut ids that have been registered mapped to user-readable descriptions for the app to present that describes the shortcut.
+            </para></listitem>
```

Comment:
as a further thought to future-proof this: maybe `aa{sv}`, so an array of dicts, resolving to something like:
```yaml
- shortcut_id: foo
  description: blah
  enabled: true
- shortcut_id: bar
  ...
```

Reply:
There is also the risk of overcomplicating everything just in case.

Reply:
How do you mean the information should be listed if the type is `a{sv}`? If the key is the `id`, then what`s the `v`?

Reply:
This has been addressed since